### DISCUSSION TITLE
[VTS] Liveness score is not as precise as we would like to

### DISCUSSION THREAD

**Discussion by onmax** [nimiq/developer-center#23] [General]
The current specification only takes two states into account when calculating the liveness score: ~~producing blocks and not producing blocks.~~ offline and online

This is counterintuitive, since the metric we want to measure is the _uptime_ of the server. Since there is no way of knowing whether a server is online or not, the only thing we can know is whether the validator has produced blocks when it should have done so.

If we had a way to know exactly whether validators were online/offline, given a timestamp, a more accurate "liveness" score could be used, and we could even use the name "uptime" instead of "liveness".

We have already discussed this as a team and decided to move forward with this decision. But we decided to leave it here in the discussion for reference.

Small note: You could argue that "uptime" and "reliability" measure the same thing, but that's not quite true:
- Liveness: To know whether the validator is online most of the time, otherwise it will be penalised. 
- Reliability: to know if the validator behaves well once selected.

**Comment by sisou** [comment]
So this is confusing. Checking for block production is for `reliability`. The documentation says that `liveness` only comes from being elected vs not elected, nothing about producing blocks. So which one is it, e.g. in the heading of this thread?
